### PR TITLE
Improves the CI tests for reporting

### PR DESCRIPTION
**Pull request by sergiorg-hpc**
**Description**

- [x] Added reference files for better report comparison.
- [x] Renamed cmake variables to increase readability.

**How to test this?**

```bash
module load cmake gcc hpe-mpi reportinglib 
cmake -DCORENRN_ENABLE_REPORTING=ON -DCMAKE_PREFIX_PATH="/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/intel-19.1.2.254/libsonata-report-0.1.5-zc4zgl" ..
make -j8
ctest -V -R reporting_1
```

**Test System**
 - OS: RHE Linux BB5
 - Compiler: gcc/9.3.0
 - Version: master
 - Backend: CPU

**Use certain branches for the SimulationStack CI**

CI_BRANCHES:NEURON_BRANCH=master,

### CODE REVIEW

File: tests/integration/negative_integration_test.sh.in [hunk 1]
```diff
@@ -25,7 +25,7 @@ fi
 cd @CMAKE_CURRENT_BINARY_DIR@/@SIM_NAME@
 
 # We convert spikes to out.dat format
-reports=@ENABLE_SONATA_REPORTS@
+reports=@ENABLE_SONATA_REPORTS_TESTS@
 if [ "$reports" = "ON" ]
 then
   data=$(@H5DUMP_EXECUTABLE@ -d /spikes/timestamps -d /spikes/node_ids -y -O spikes | tr "," "\n")
```
Comment (iomaganaris) [inline tests/integration/negative_integration_test.sh.in:31]: I think there is an issue with the datasets' names that fails the CI. It should be I believe:
```suggestion
  data=$(@H5DUMP_EXECUTABLE@ -d /spikes/All/timestamps -d /spikes/All/node_ids -y -O spikes | tr "," "\n")
  ```

**Review by iomaganaris — COMMENTED** [review]
The changes LGTM! Much clearer. I just have a suggestion to fix the failing CI

File: tests/integration/negative_integration_test.sh.in (hunk 1 above)
Comment (jorblancoa) [reply tests/integration/negative_integration_test.sh.in:31]: Yes! we are aware and working on it, this is the first issue that after fixing it we see new ones :D

**Comment by jorblancoa** [comment]
retest this please jenkins

**Comment by jorblancoa** [comment]
retest this please

**Review by iomaganaris — APPROVED** [review]

**Pull request merged**